ui/components/extensions/adapters/adapters.js

```diff
@@ -0,0 +1,175 @@
+import { useEffect, useState } from "react";
+import { withRouter } from "next/router";
+import { iconMedium, extensionStyles as styles } from "../../../css/icons.styles";
+import { Grid, Typography, IconButton, Switch, CloseIcon } from "@material-ui/core";
+import { withStyles } from "@material-ui/core/styles";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { withSnackbar } from "notistack";
+import { updateProgress } from "../../../lib/store";
+import { INITIAL_GRID_SIZE } from "../../../utils/Enum";
+import { adaptersList } from "./constants";
+import changeAdapterState from '../../graphql/mutations/AdapterStatusMutation';
+
+const Adapters = ({ updateProgress, enqueueSnackbar, closeSnackbar, classes, meshAdapters, meshAdaptersts }) => {
+
+  // States.
+  const [availableAdapters, setAvailableAdapters] = useState(adaptersList);
+  const [mts, setMts] = useState(new Date());
+
+  // useEffects.
+  // TODO: This way of updating progress is really bad. Need to make this more elegant.
```

Comment:
suggestions?

Reply:
Well as of now I am thinking about react `Suspense`. What do you think about it?

Reply:
Good suggestion for this usecase, but I think its not the effort to make right now, unless UI is behaving completely bad under this. 

The react-skeletons can be used for loading components, which I think should be a good improvement on catalog, designs, and applications page.

Also, if you are good, please open a PR with the POC or prototype for the react-suspense, implementing it at only one place to show how its different.

Reply:
Yes a POC for sure makes sense. cool then I'll do a POC for this.